packages/image_picker/ios/Classes/ImagePickerPlugin.m

```diff
@@ -118,6 +119,42 @@ - (void)showCamera {
   }
 }
 
+- (void)checkAuthorization {
+  AVAuthorizationStatus status = [AVCaptureDevice authorizationStatusForMediaType:AVMediaTypeVideo];
+  
+  switch (status) {
+    case AVAuthorizationStatusAuthorized:
+      [self showCamera];
+      break;
+    case AVAuthorizationStatusNotDetermined: {
+      [AVCaptureDevice requestAccessForMediaType:AVMediaTypeVideo completionHandler:^(BOOL granted) {
+        if (granted) {
+          dispatch_async(dispatch_get_main_queue(), ^{
+            if (granted) {
+              [self showCamera];
+            }
+          });
+        } else {
+          dispatch_async(dispatch_get_main_queue(), ^{
+            [self errorNoAccess];
+          });
+        }
+      }];
+    }; break;
+    case AVAuthorizationStatusDenied:
```

Comment:
It is prob a good idea to send the failed statuses to the dart code so the developers can have more details on what caused the failure.